joinservice/internal/kubernetes/kubernetes.go

```diff
@@ -85,7 +86,11 @@ func (c *Client) CreateConfigMap(ctx context.Context, configMap corev1.ConfigMap
 func (c *Client) AddNodeToJoiningNodes(ctx context.Context, nodeName string, componentsHash string, isControlPlane bool) error {
 	joiningNode := &unstructured.Unstructured{}
 
-	objectMetadataName := nodeName
+	compliantNodeName := k8sCompliantHostname(nodeName)
+
+	// JoiningNodes referencing a worker node are named after the worker node.
+	// JoiningNodes referencing the control-plane node are named "control-plane".
```

Comment:
I don't understand what this comment refers to. The line below seems somewhat unrelated, we always assign the same compliant name, no matter if this is a control plane or not.

Reply:
The comment refers to the whole code block. The first line explains the first two lines and the second line refers to the `if`. I wanted to keep the comment together as to highlight the difference.